proxy/envoy/config.go

```diff
@@ -227,6 +226,61 @@ func buildOutboundListeners(instances []*model.ServiceInstance, services []*mode
 	return listeners, clusters
 }
 
+// buildDestinationHTTPRoutes creates HTTP route for a service and a port from rules
+func buildDestinationHTTPRoutes(service *model.Service,
+	servicePort *model.Port,
+	rules []*proxyconfig.RouteRule) []*HTTPRoute {
+	protocol := servicePort.Protocol
+	switch protocol {
+	case model.ProtocolHTTP, model.ProtocolHTTP2, model.ProtocolGRPC:
+		routes := make([]*HTTPRoute, 0)
+
+		// collect route rules
+		useDefaultRoute := true
+		for _, rule := range rules {
+			if rule.Destination == service.Hostname {
+				httpRoute := buildHTTPRoute(rule, servicePort)
+				routes = append(routes, httpRoute)
+
+				// User can provide timeout/retry policies without any match condition,
+				// or specific route. User could also provide a single default route, in
+				// which case, we should not be generating another default route.
+				// For every HTTPRoute we build, the return value also provides a boolean
+				// "catchAll" flag indicating if the route that was built was a catch all route.
+				// When such a route is encountered, we stop building further routes for the
+				// destination and we will not add the default route after the for loop.
+				if httpRoute.CatchAll() {
+					useDefaultRoute = false
+					break
+				}
+			}
+		}
+
+		if useDefaultRoute {
```

Comment:
I think we also need to add rules with a single route (same destination, 100 weight) as a catch all route.
examples of catch all routes:
1. rule which has prefix /, no http headers and no path
2. rule which has no match, with single route (weight 100)
3. any thing else @frankbu ??